google-cloud-spanner/lib/google/cloud/spanner/errors.rb

```diff
@@ -20,6 +20,16 @@ module Cloud
     module Spanner
       ##
       # @private
+      # # CommitError
+      #
+      # Signal that the Transaction has been committed.
```

Comment:
What about copying Rails' ActiveRecord Transaction flow control? In ActiveRecord (afaik), unless an exception is thrown, any return (early return, break, returning a falsy value, whatever) will commit the transaction. Only exceptions result in rollback. A special [`ActiveRecord::Rollback`](http://api.rubyonrails.org/classes/ActiveRecord/Rollback.html) error type is provided to users for triggering rollback. It is caught by the framework and does not result in an application error. (See link above for example code.)